compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -1135,7 +1135,7 @@ impl<'a> Linker for EmLinker<'a> {
         self.cmd.arg(match self.sess.opts.debuginfo {
             DebugInfo::None => "-g0",
             DebugInfo::Limited => "-g3",
-            DebugInfo::Full => "-g4",
+            DebugInfo::Full => "-gsource-map",
```

Comment:
The "fullest" debug info is DWARF, which is just `-g`.

So I would suggest `-g` for full and `--profiling-funcs` for limited.   (Or `-gsource-map`  for limited if you prefer).

Reply:
How's it look now?